internal/tokenizer/flex/tokenize_c.go

```diff
@@ -0,0 +1,71 @@
+package flex
+
+// #include <stdlib.h>
+// #include "linguist.h"
+// #include "lex.linguist_yy.h"
+// int linguist_yywrap(yyscan_t yyscanner) {
+// 	return 1;
+// }
+import "C"
+import "unsafe"
+
+const maxTokenLen = 32 // bytes
+
+// TokenizeFlex implements tokenizer by calling Flex generated code from linguist in C
+// This is a transliteration from C https://github.com/github/linguist/blob/master/ext/linguist/linguist.c#L12
+func TokenizeFlex(content []byte) []string {
+	var buf C.YY_BUFFER_STATE
+	var scanner C.yyscan_t
+	var extra C.struct_tokenizer_extra
+	var _len C.ulong
+	var r C.int
+
+	_len = C.ulong(len(content))
+	cs := C.CBytes(content)
+	defer C.free(unsafe.Pointer(cs))
+
+	C.linguist_yylex_init_extra(&extra, &scanner)
+	buf = C.linguist_yy_scan_bytes((*C.char)(cs), _len, scanner)
+
+	ary := []string{}
```

Comment:
```suggestion
	var ary []string
```
[source](https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices)

Reply:
This would result in returning `nil` instead of empty slice and that would be inconsistent with current impl of tokenizer API, so I think it's better to keep it as is.

Reply:
I'm not sure if it matters from the API point of view - it's pretty rare to test for slice nil-ness.